codegen/vulkan/generate_structure_type_names.py

```diff
@@ -0,0 +1,147 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import argparse
+from common import print_vulkan_helper, StructMembers, gen_hpp_header
+from helper_generator import HelperOutputGenerator
+
+try:
+    import reg
+    from vkconventions import VulkanConventions
+except ModuleNotFoundError:
+    print_vulkan_helper()
+
+from lxml import etree
+
+
+class StructureTypeNameGenerator(HelperOutputGenerator):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    lastUsedFeature = None
+
+    def beginFile(self, genOpts):
+        super().beginFile(genOpts)
+
+        header = gen_hpp_header() + \
+            '#include <volk.h>\n' + \
+            '\n' + \
+            'namespace components\n' + \
+            '{\n' + \
+            'namespace vulkan\n' + \
+            '{\n' + \
+            'template <typename Type>\n' + \
+            'VkStructureType get_structure_type_name()\n' + \
+            '{\n' + \
+            '	throw "function not implemented";\n' + \
+            '}\n'
+
+        self.outFile.write(header)
+
+    def endFile(self):
+
+        self.outFile.write(
+            '#endif' +  # end last feature
+            '}        // namespace vulkan\n' +
+            '}        // namespace components\n')
+
+        super().endFile()
+
+    def enterStruct(self, name: str, info: StructMembers, is_alias: bool) -> str:
+        members = info.get_members()
+
+        if is_alias:
+            return None
+
+        # A new feature is either the first feature of a set...
+        is_first_feature = self.lastUsedFeature == None
+        is_new_feature = is_first_feature
+        nextFeatureName = self.featureName
+
+        # ... Or the first change between feature sets
+        if self.featureName is not self.lastUsedFeature:
+            is_new_feature = True
+
+        for member in members:
+            if member.name == "sType":
+                structureName = member.elem.get("values")
+
+                section = ""
+
+                # end previous feature
+                if is_new_feature:
+                    if not is_first_feature:
+                        # first endif isn't required
+                        section += f'#endif\n\n'
+
+                    section += f'#ifdef {nextFeatureName}\n'
+
+                    # Crucial: do not remove
+                    self.lastUsedFeature = self.featureName
+
+                if structureName:
+                    section += 'template <>\n' + \
+                        f'VkStructureType get_structure_type_name<{name}>()\n' + \
```

Comment:
With `get_structure_type_name`, I would expect to get some string, or so.
Why not just name it `get_structure_type`?

Reply:
I'll make this change as soon as I can :)